### ISSUE TITLE
Handle json deserialization exception

### ISSUE THREAD

**Issue by maheshrajamani** [stargate/jsonapi#56]
Create exception mapper for JsonMappingException

**Comment by tatu-at-datastax** [comment]
@ivansenic Jackson is quite permissive when target type is `String`: this conversion is indeed allowed and automatic by default. Same would be true for numbers.
(original rationale -- this is from Jackson 1.0 or before -- is to try support coercions that are safe -- there are users who really like looseness, and others who dislike it, difficult to find defaults everyone agrees with).

There are ways to configure stricter handling (added in 2.14.x) if we feel this is important to prevent, see:

* https://github.com/FasterXML/jackson-databind/issues/3013 (for integer->String)
* https://github.com/FasterXML/jackson-databind/issues/3613 (for boolean->String)

(default behavior cannot be changed due to backwards-compatibility constraints)

**Comment by tatu-at-datastax** [comment]
Change to `ObjectMapper` building would be something like (from PR test above):

```
   ObjectMapper mapper = JsonMapper.builder()
           // For logical target type of "Textual" (includes String)
            .withCoercionConfig(LogicalType.Textual, cfg ->
                    // fail conversion from source Shape of Boolean
                    cfg.setCoercion(CoercionInputShape.Boolean, CoercionAction.Fail))
            .build();
```

**Comment by ivansenic** [comment]
@tatu-at-datastax I merged #119. If you think we need more stricter handling please open a new issue.

**Comment by tatu-at-datastax** [comment]
> @tatu-at-datastax I merged #119. If you think we need more stricter handling please open a new issue.

Thank you! I think we are good now (I don't see an immediate problem); will keep this in mind should there be something to make it an issue.
